### ISSUE TITLE
Windows build machine cannot produce **any** working builds

### ISSUE THREAD

**Issue by zaneclaes** [bazelbuild/rules_docker#513]
**tl;dr**: it seems that `guess_runfiles` (which is copy-pasta-ed all over the repo) makes invalid assumptions about the output directory.

Perhaps I'm missing something in the docs, but it appears that the `[OUT].runfiles` directory yields only a `manifest` file, which subsequently causes anything after the `build` image step to fail. Note that everything tried here works fine on OSX. And the `build` step succeeds; I just can't `run` to turn anything into an image (or tag it, for that matter).

Assuming a simple container `BUILD` (n.b., `@filebeat` is just a pull of `docker.elastic.co/beats/filebeat:6.4.0`).

```
container_image(
  name = "DEV",
  base = "@filebeat//image",
)
```

Attempt 1: Running Directly
----
`bazel run //service/logpipe:DEV`

```
INFO: Analysed target //service/logpipe:DEV (25 packages loaded).
INFO: Found 1 target...
Target //service/logpipe:DEV up-to-date:
  C:/users/zane/_bazel_zaneclaes/eojxvnhj/execroot/__main__/bazel-out/x64_windows-fastbuild/bin/service/logpipe/DEV-layer.tar
INFO: Elapsed time: 306.810s, Critical Path: 13.48s
INFO: 21 processes: 21 local.
INFO: Build completed successfully, 43 total actions
INFO: Running command line: C:/users/zane/_bazel_zaneclaes/eojxvnhj/execroot/__main__/bazel-out/x64_windows-fastbuildINFO: Build completed successfully, 43 total actions
/bin/bash: C:/users/zane/_bazel_zaneclaes/eojxvnhj/execroot/__main__/bazel-out/x64_windows-fastbuild/bin/service/logpipe/DEV: No such file or directory
```

If I run the actual output script (`DEV`) from the shell it complains more explicitly about the files missing from the `DEV.runfiles`. Inspecting this directory, I see only the `MANIFEST` file.

Attempt 2: Exporting a Tar
----
`bazel run //service/logpipe:DEV.tar`

```
INFO: Analysed target //service/logpipe:DEV.tar (0 packages loaded).
INFO: Found 1 target...
Target //service/logpipe:DEV.tar up-to-date:
  C:/users/zane/_bazel_zaneclaes/eojxvnhj/execroot/__main__/bazel-out/x64_windows-fastbuild/bin/service/logpipe/DEV.tar
INFO: Elapsed time: 13.026s, Critical Path: 11.90s
INFO: 12 processes: 12 local.
INFO: Build completed successfully, 15 total actions
```

Oh hey! Seems good, right? Not quite. Let's try:
```
docker import -i C:/users/zane/_bazel_zaneclaes/eojxvnhj/execroot/__main__/bazel-out/x64_windows-fastbuild/bin/service/logpipe/DEV.tar
```

The tar it created is invalid:
```
1d31b5806ba4: Loading layer [==================================================>]    131kB/131kB
Error processing tar file(exit status 1): unexpected EOF
```

Attempt 3: Bundle It!
----

Tried using `container_bundle`. Same problems.

Attempt 4: Different Image
----

Also tried `mysql/mysql-server`, `ubuntu`, and a dozen others.

Attempts 5-100:
----

I've been at this for over 30 hours now. Nothing seems to work.

**Comment by nlopezgi** [comment]
Thanks for reporting this issue, and thanks for trying out rules_docker! Sorry to hear about these issues. We do not actively test rules_docker with windows (although the documentation states its supported, apologies for the misleading statement). However, I have seen other issues reported by Windows users (e.g., https://github.com/bazelbuild/rules_docker/issues/507, https://github.com/bazelbuild/rules_docker/issues/362) in which our official statement is that xx_image rules are not supported on windows, but the container_image _should_ work. Its possible however, that given we do not test this regularly, there has been some change in upstream Bazel related to Windows (I know there's been lots of activity around better support for Bazel on Windows) which might have resulted in the issues you are seeing. Though I must say I'm a bit surprised as https://github.com/bazelbuild/rules_docker/issues/507 is a recent issue - @enriched (since you opened the issue mentioned) could you comment on whether you can build containers on Windows and any details about your setup that could help @zaneclaes? 
@zaneclaes could you provide a bit more info about which version of Bazel and which version of these rules you are using (also anything relevant about Windows version)?

**Comment by enriched** [comment]
@nlopezgi @zaneclaes Sorry, I still haven't been able to get the `container_image` rule to work on windows. I have some other problems(unrelated to rules_docker) that I am working through to get our build working on windows.

I was able to get a bit further with these rules on windows after I installed gzip on the path (`choco install gzip`). But I don't think that you are seeing the same error I was seeing so I assume that you have that installed.

**Comment by zaneclaes** [comment]
I never had any problems with zip, though I'm not using it to my knowledge (unless its under-the-hood somewhere). In any case, I rewrote the templates in this PR so that it works now. https://github.com/bazelbuild/rules_docker/pull/514

**Comment by nlopezgi** [comment]
Thanks for your comments @enriched 
Responded on the PR, but also want to leave a response in this issue: 
Currently rules_docker has deep issues that prevent it from working on windows.
However, there is upcoming work by the Bazel team that will help make this easier to fix (https://github.com/bazelbuild/bazel/issues/4319 - https://docs.google.com/document/d/e/2PACX-1vRCSB_n3vctL6bKiPkIa_RN_ybzoAccSe0ic8mxdFNZGNBJ3QGhcKjsL7YKf-ngVyjRZwCmhi_5KhcX/pub). I do have plans to have our team do some of the work (related to how tools and bash is being used in the rules) which is a prerequisite for these rules to work on windows (planned for the upcoming weeks). I will keep you informed. At the moment, however, the changes in https://github.com/bazelbuild/rules_docker/pull/514 are not the right solution to support windows builds. I will remove the statement regarding Windows support from the docs and replace it with a note regarding our upcoming work to get this supported properly.
